src/services/upload.ts

```diff
@@ -0,0 +1,125 @@
+import { storage } from "~/firebase/init";
+import {
+  ref,
+  uploadBytes,
+  getDownloadURL,
+  FirebaseStorage,
+  listAll,
+  StorageReference,
+  uploadBytesResumable,
+  deleteObject,
+} from "firebase/storage";
+import { Dispatch } from "react";
+
+type UploadFilePath = {
+  userId: string;
+  file: File;
+};
+type UpdateFile = {
+  filePath: string;
+  file: File;
+};
+
+class UploadService {
+  public readonly rootFolder: string = "jazzapp/music/";
+
+  public async uploadFile(
+    fPath: UploadFilePath,
+    setProgress?: Dispatch<{ percents: number; operation: string }>,
+  ): Promise<{ status: boolean; url?: string }> {
+    if (!fPath.file) return Promise.resolve({ status: false });
+
+    const subMusicRef = `/${fPath.userId}/${fPath.file.name}`;
+    const fullMusicRef = ref(storage, `${this.rootFolder}${subMusicRef}`);
+    const uploadTask = uploadBytesResumable(fullMusicRef, fPath.file);
+
+    uploadTask.on(
+      "state_changed",
+      (snapshot) => {
+        const progress = Math.round(
+          (snapshot.bytesTransferred / snapshot.totalBytes) * 100,
+        );
+        if (setProgress) {
+          setProgress({ percents: progress, operation: "Upload" });
+        }
+      },
+      (error) => {
+        console.error(error);
+        Promise.reject({ status: false });
+      },
+    );
+    await uploadTask;
+
+    const url = await getDownloadURL(fullMusicRef);
+
+    return Promise.resolve({ status: true, url });
+  }
+  public deleteFile(filePath: string): Promise<{ status: boolean }> {
+    const deleteRef = ref(storage, `${this.rootFolder}${filePath}`);
+    deleteObject(deleteRef)
+      .then(() => {
+        console.log(`File deleted: ${deleteRef}`);
+      })
+      .catch((error) => {
+        console.error(deleteRef.fullPath);
+        console.error(error);
+        return Promise.resolve({ status: false });
+      });
+    return Promise.resolve({ status: true });
+  }
+  public async updateFile(
+    fPath: UpdateFile,
```

Comment:
Same as above, try to split this into primitive arguments.
```suggestion
    filePath: string,
    file: File,
```